queries/java/highlights.scm

```diff
@@ -2,20 +2,33 @@
 
 ; Methods
 
+[
 (method_declaration
-  name: (identifier) @method)
+ name: (identifier))
 (method_invocation
-  name: (identifier) @method)
+ name: (identifier))
+] @method
+
+[
+(method_declaration
+ parameters: (formal_parameter))
+(method_declaration
+ parameters: (formal_parameter))
+] @parameter
```

Comment:
Could you check whether you're also capturing lambda parameters? Probably, this requires an additional query.

Reply:
Also check the parameters of constructors:

![image](https://user-images.githubusercontent.com/7189118/85225925-c6532d80-b3d4-11ea-8248-0a97b46547e4.png)

Reply:
I guess: this should be 

```
(method_declaration
   parameters: (formal_parameter) @parameter)
```
Or it will capture the whole method as `@parameter`.

![image](https://user-images.githubusercontent.com/7189118/85226072-cf90ca00-b3d5-11ea-8035-297590bf7da9.png)

Reply:
> Could you check whether you're also capturing lambda parameters? Probably, this requires an additional query.

Idk what are lamda parameters

Reply:
> I guess: this should be
> 
> ```
> (method_declaration
>    parameters: (formal_parameter) @parameter)
> ```
> 
> Or it will capture the whole method as `@parameter`.
> 
> ![image](https://user-images.githubusercontent.com/7189118/85226072-cf90ca00-b3d5-11ea-8035-297590bf7da9.png)

Isn't it like this already?

Reply:
> Idk what are lamda parameters

I think he meant lambda function (or closure) parameters

Reply:
> > I guess: this should be
> > ```
> > (method_declaration
> >    parameters: (formal_parameter) @parameter)
> > ```
> > 
> > 
> > Or it will capture the whole method as `@parameter`.
> > ![image](https://user-images.githubusercontent.com/7189118/85226072-cf90ca00-b3d5-11ea-8035-297590bf7da9.png)
> 
> Isn't it like this already?

No, your query doesn't match anything. I guess you want something like:

```

(method_declaration
 parameters: 
 	(formal_parameters
 		(formal_parameter
        	name: (identifier) @parameter))) 
```
Or simpler:
```
(formal_parameter
        	name: (identifier) @parameter))) 
```
Or:
```
(formal_parameter) @parameter 
```

Reply:
You can test your queries here: https://tree-sitter.github.io/tree-sitter/playground

Reply:
> ```
> (formal_parameter
>         	name: (identifier) @parameter))) 
> ```

I suggest using this one, as you capture the actual thing you want to capture